### PR TITLE
IGNITE-19531 Switch table IDs from UUIDs to integers

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-19531

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java [hunk 1]
```diff
@@ -909,6 +924,24 @@ public int readInt() {
         return val;
     }
 
+    /** {@inheritDoc} */
```
Comment (tkalkirill) [inline modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java:927]:
```suggestion
```

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/PlannerTest.java [hunk 2]
```diff
@@ -192,6 +192,10 @@ public ColocationGroup colocationGroup(MappingQueryContext ctx) {
         assertEquals(2, plan.fragments().size());
     }
 
+    private static int randomTableId() {
+        return new Random().nextInt();
```
Comment (tkalkirill) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/PlannerTest.java:196]: why not 1 ?)

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/TableSpoolPlannerTest.java [hunk 3]
```diff
@@ -82,4 +83,8 @@ public void tableSpoolDistributed() throws Exception {
 
         checkSplitAndSerialization(phys, publicSchema);
     }
+
+    private static IgniteDistribution affinity() {
+        return IgniteDistributions.affinity(0, new Random().nextInt(), DEFAULT_ZONE_ID);
```
Comment (tkalkirill) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/TableSpoolPlannerTest.java:88]: why not 1 ?)

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/UnionPlannerTest.java [hunk 4]
```diff
@@ -109,9 +109,13 @@ private IgniteSchema prepareSchema() {
...
                         .build(),
-                IgniteDistributions.affinity(0, UUID.randomUUID(), DEFAULT_ZONE_ID)
+                IgniteDistributions.affinity(0, randomTableId(), DEFAULT_ZONE_ID)
         );
 
         return publicSchema;
     }
+
+    private static int randomTableId() {
+        return new Random().nextInt();
```
Comment (tkalkirill) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/UnionPlannerTest.java:119]: why not 1 ?)

File: modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/HashIndexDescriptor.java [hunk 5]
```diff
@@ -151,6 +153,17 @@ private static List<HashIndexColumnDescriptor> extractIndexColumnsConfiguration(
                 .collect(toUnmodifiableList());
     }
 
+    @Nullable
+    private static TableView findTableById(int tableId, NamedListView<? extends TableView> tablesView) {
```
Comment (tkalkirill) [inline modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/HashIndexDescriptor.java:157]:
```suggestion
    private static @Nullable TableView findTableById(int tableId, NamedListView<? extends TableView> tablesView) {
```

**Comment by ptupitsyn** [comment]
Client-related changes (Java, C#) look good to me.

**Review by ptupitsyn — APPROVED** [review]

**Comment by arcolight** [comment]
C++ changes LGTM.

File: modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java [hunk 6]
```diff
@@ -45,8 +46,8 @@ public class IndexValidatorImplTest {
     private TablesConfiguration tablesConfig;
 
     @BeforeEach
-    void setupTableConfig() {
-        tablesConfig.tables().get("fooTable").id().update(TABLE_ID);
+    void setupTableConfig() throws Exception {
+        tablesConfig.tables().get("fooTable").id().update(TABLE_ID).get(1, TimeUnit.SECONDS);
```
Comment (tkalkirill) [inline modules/schema/src/test/java/org/apache/ignite/internal/schema/configuration/index/IndexValidatorImplTest.java:50]: Please use `org.apache.ignite.internal.testframework.matchers.CompletableFutureMatcher#willSucceedFast` or similar

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/HashIndexPlannerTest.java [hunk 7]
```diff
@@ -64,13 +65,17 @@ public void hashIndexIsAppliedForEquiCondition() throws Exception {
         assertThat(invalidPlanMsg, scan.indexName(), equalTo(indexName));
     }
 
+    private static IgniteDistribution affinity() {
+        return IgniteDistributions.affinity(0, new Random().nextInt(), DEFAULT_ZONE_ID);
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/HashIndexPlannerTest.java]: Originally there was a random table ID, so it makes sense to make sure that the table IDs are different at each call. Random IDs don't guarantee uniqueness, so I changed it to using a global counter.

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**